application/ui/components/param-row.jsx

```diff
@@ -0,0 +1,221 @@
+                name      = {'include-' + component.props.param.name}
+                onChange  = {component.props.onInclude}
+                checked   = {component.props.isIncluded}
+            />
+        );
+        if (component.props.param.type !== 'file') {
+            nullCheckboxComponent = (
+                <input
+                    type = "checkbox"
+                    name = {'null-' + component.props.param.name}
+                    onChange  = {component.props.onNull}
+                    checked   = {component.props.isNull}
+                />
+            );
+        } else {
+            nullCheckboxComponent = ('N/A');
+        }
+
+        switch (component.props.param.type) {
+            // row-params
+            case 'hash':
+            case 'object':
+                if (!Params) {
+                    Params = require('./params');
+                }
+                rowParam = (
+                    <tr>
+                        <td colSpan={5}>
+                            <Params
+                                key={component.props.key}
+                                params={component.props.param.params}
+                                onChange={component.props.onChange}
+                            />
+                        </td>
+                    </tr>
+                );
+                break;
+            case 'array':
+                rowParam = (
+                    <tr>
+                        <td colSpan={5}>
+                            <ArrayObject
+                                key={component.props.key}
+                                param={component.props.param.param}
+                                onChange={component.props.onChange}
+                            />
+                        </td>
+                    </tr>
+                );
+                break;
+            case 'custom-object':
+                rowParam = (
+                    <tr>
+                        <td colSpan={5}>
+                            <CustomObject name='' onChange={component.props.onChange} />
+                        </td>
+                    </tr>
+                );
+                break;
+
+            // inline params
+            case 'enum':
+                var enumValues = _.map(component.props.param.enumValues, function (value) {
+                    return {
+                        label : value,
+                        value : value
+                    };
+                });
+                inlineParam = (
+                    <Select value={component.props.value} key={component.props.key} options={enumValues} onChange={component.props.onChange} />
+                );
+                break;
+            case 'boolean':
+                inlineParam = (
+                    <Select value={component.props.value} key={component.props.key} options={[{value: false, label: 'false'}, {value: true, label: 'true'}]} onChange={component.props.onChange} />
```

Comment:
This rendered component (and the one above it) could be a bit more easy to read by splitting it out into multiple lines.